Descent3/LoadLevel.cpp

```diff
@@ -3588,6 +3588,86 @@ static inline char* GetCurrentSumString() {
   return output_buf;
 }
 
+bool GetLevelInfo(const std::filesystem::path &filename, level_info &info) {
+  CFILE *ifile = cfopen(filename, "rb");
+  bool found = false;
+
+  if (!ifile) {
+    LOG_ERROR.printf("Failed to open mission file %s", filename.u8string().c_str());
+    return false;
+  }
+
+  try {
+    // catch cfile errors
+
+    // Read & check tag
+    char tag[4];
+    cf_ReadBytes((uint8_t *)tag, 4, ifile);
+    if (strncmp(tag, LEVEL_FILE_TAG, 4)) {
```

Comment:
can be extracted to a new function. This operation is also done for `LoadLevel`. Same for level file version check and initialization of level info